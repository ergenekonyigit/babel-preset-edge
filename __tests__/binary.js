import buildPreset from "../src"
import { fixtures, titles, check } from "./core"

const options = buildPreset(null, { target: "binary", sourceMaps: false, env: "production" })

fixtures.forEach((fileName, index) => {
  test(titles[index], () => check(fileName, options))
})
