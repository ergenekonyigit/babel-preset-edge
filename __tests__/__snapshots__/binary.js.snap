// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comments 1`] = `
"\\"use strict\\";

var _camelCase2 = require(\\"lodash/camelCase\\");

var _camelCase3 = _interopRequireDefault(_camelCase2);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { \\"default\\": obj }; }

function translate(value) {
  return (0, _camelCase3[\\"default\\"])(value);
}"
`;

exports[`deadcode 1`] = `
"\\"use strict\\";

const STATIC = false;"
`;

exports[`es3: literal 1`] = `
"\\"use strict\\";

foo[\\"catch\\"];"
`;

exports[`es3: member 1`] = `
"\\"use strict\\";

var foo = {
  \\"catch\\": function () {}
};"
`;

exports[`es2015: classes 1`] = `
"\\"use strict\\";

exports.__esModule = true;
class Component {
  main() {}
}
exports.Component = Component;"
`;

exports[`es2015: generator 1`] = `
"\\"use strict\\";

function* idMaker() {
  var index = 0;
  while (true) yield index++;
}"
`;

exports[`es2015: set 1`] = `
"\\"use strict\\";

new Set([1, 2, 3]);"
`;

exports[`es2016: exponentiation 1`] = `
"\\"use strict\\";

var result = Math.pow(2, 3);"
`;

exports[`esnext: async 1`] = `
"\\"use strict\\";

var _promise = require(\\"nodent-runtime/promise\\");

var _promise2 = _interopRequireDefault(_promise);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { \\"default\\": obj }; }

function wait() {
  return new Promise(($return, $error) => loadData().then(function ($await_1) {
    return $return();
  }.$asyncbind(this, $error), $error));
}"
`;

exports[`esnext: class properties 1`] = `
"\\"use strict\\";

exports.__esModule = true;
class Component {
  constructor() {
    this.onClick = () => {};
  }

}
exports.Component = Component;"
`;

exports[`esnext: dynamic import 1`] = `
"\\"use strict\\";

Promise.resolve().then(() => require(\\"lodash\\")).then(lodash => {
  console.log(lodash.camelCase(\\"hello world\\"));
});"
`;

exports[`esnext: import 1`] = `
"\\"use strict\\";

var _camelCase2 = require(\\"lodash/camelCase\\");

var _camelCase3 = _interopRequireDefault(_camelCase2);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { \\"default\\": obj }; }

console.log((0, _camelCase3[\\"default\\"])(\\"hello world\\"));"
`;

exports[`esnext: object spread 1`] = `
"\\"use strict\\";

let original = { bar: 10 };
let variant = Object.assign({ foo: 1 }, original);"
`;

exports[`lodash: async 1`] = `
"\\"use strict\\";

var _parallel2 = require(\\"async/parallel\\");

var _parallel3 = _interopRequireDefault(_parallel2);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { \\"default\\": obj }; }

(0, _parallel3[\\"default\\"])([]);"
`;

exports[`lodash: lodash 1`] = `
"\\"use strict\\";

var _camelCase2 = require(\\"lodash/camelCase\\");

var _camelCase3 = _interopRequireDefault(_camelCase2);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { \\"default\\": obj }; }

(0, _camelCase3[\\"default\\"])(\\"hello world\\");"
`;

exports[`lodash: recompose 1`] = `
"\\"use strict\\";

var _compose2 = require(\\"recompose/compose\\");

var _compose3 = _interopRequireDefault(_compose2);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { \\"default\\": obj }; }

(0, _compose3[\\"default\\"])([]);"
`;

exports[`react: jsx 1`] = `
"\\"use strict\\";

var _jsx2 = require(\\"babel-runtime/helpers/jsx\\");

var _jsx3 = _interopRequireDefault(_jsx2);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { \\"default\\": obj }; }

function render() {
  return (0, _jsx3[\\"default\\"])(\\"h1\\", {}, void 0, \\"Hello\\");
}"
`;

exports[`react: jsx spread 1`] = `
"\\"use strict\\";

function render(props) {
  return React.createElement(
    \\"h1\\",
    Object.assign({ className: \\"large\\" }, props),
    \\"Hello\\"
  );
}"
`;

exports[`react: proptypes 1`] = `
"\\"use strict\\";

var _jsx2 = require(\\"babel-runtime/helpers/jsx\\");

var _jsx3 = _interopRequireDefault(_jsx2);

var _react = require(\\"react\\");

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { \\"default\\": obj }; }

class MyComponent extends _react2[\\"default\\"].Component {
  render() {
    return (0, _jsx3[\\"default\\"])(\\"div\\", {}, void 0, this.props.userName);
  }
}"
`;
